com.unity.render-pipelines.universal/Shaders/Terrain/TerrainLitPasses.hlsl

```diff
@@ -47,16 +47,20 @@ struct Varyings
 #endif
 
 #if defined(_NORMALMAP) && !defined(ENABLE_TERRAIN_PERPIXEL_NORMAL)
-    float4 normal                   : TEXCOORD3;    // xyz: normal, w: viewDir.x
-    float4 tangent                  : TEXCOORD4;    // xyz: tangent, w: viewDir.y
-    float4 bitangent                : TEXCOORD5;    // xyz: bitangent, w: viewDir.z
+    half4 normal                    : TEXCOORD3;    // xyz: normal, w: viewDir.x
+    half4 tangent                   : TEXCOORD4;    // xyz: tangent, w: viewDir.y
+    half4 bitangent                 : TEXCOORD5;    // xyz: bitangent, w: viewDir.z
 #else
-    float3 normal                   : TEXCOORD3;
-    float3 viewDir                  : TEXCOORD4;
+    half3 normal                    : TEXCOORD3;
+    half3 viewDir                   : TEXCOORD4;
```

Comment:
Isn't it worth it to move viewDir computation to fragment like the other shaders?